### DISCUSSION TITLE
Partially selected parents

### DISCUSSION THREAD

**Discussion by amr-el-sayed** [vakata/jstree#2583] [General]
Hello!
When I use get the selection using:
the  $('#tree').jstree("get_selected", true);
the resulting array does not include the parents that have partially selected children.

Is there a way to override the prototype or events so it includes the parents node as well?
Thank you!

**Comment by vakata** [comment]
Hi, loop over ```top_selected``` and for each node invoke ```get_parents``` - merge in one and unique if needed - those are the nodes you are looking for.

**Reply by amr-el-sayed** [reply]
Hi, the prototype **get_top_selected** returns the following:
`return full ? $.map(tmp, function (i) { return this.get_node(i); }.bind(this)) : tmp;`

I can't figure out where to merge the parent.
May you show me a code example please?
